Impl: QuickBooks-Style SPA with Dashboard, Invoices, Expenses, Reconciliation, and Audit Logging

rakesh0x opened the pull request:
## Summary of Changes
- Implemented QuickBooks-style single-page application (SPA) using React + TypeScript + Vite.
- Core features implemented:
  - **Financial Dashboard**: Displays cash flow, profit & loss, and outstanding invoices.
  - **Invoice Management**: Table with filters, detail view, mark-as-paid, send reminder, edit line items.
  - **Expense Tracking**: Add expenses, categorize, and reflect totals in dashboard.
  - **Bank Reconciliation**: Stepper UI to match bank transactions with invoices/expenses and summary of unmatched items.
  - **Audit Notes Sidebar**: Logs every state mutation for RL agent verification.

- **State Management**: Implemented using `@chakra-dev/dojo-hooks` (`useDojoState` + `dojo.setState`) with seed data for offline usage.


- **Responsive Layouts**: Tested for 1280px and 1440px widths.
- **Offline Ready**: App boots fully with local seed data (`invoices`, `expenses`, `bankTransactions`).

## Screenshots / GIF

https://github.com/user-attachments/assets/0b74bc69-3289-44b8-a495-998750885789

## How to Test
1. Clone the repo and checkout this branch.
2. Install dependencies:  
   ```bash
   npm install
  
3 Run it locally 
  ```bash 
 npm run dev

rakesh0x committed:
quickbook impl

0xnirmal commented:
Quick feedback based on video:

The end deliverable needs to be as close to the real quickbooks as possible. Check out online resources to help: https://quickbooks.intuit.com/learn-support/en-us/help-article/product-setup/get-started-quickbooks-desktop/L5bYVSY73_US_en_US 

Ideally the end user cannot tell the difference between your SPA and the real thing. Thanks

rakesh0x committed:
sidebar & ui-fixes

rakesh0x committed:
lucide-icons

rakesh0x committed:
completed with nodes

rakesh0x committed:
create-invoice

rakesh0x committed:
customer-page

rakesh0x committed:
customer-page

0xnirmal commented:
hey @rakesh0x - do you mind re-recording a video demonstration?

rakesh0x commented:
yeah, ofcourse let me :)

rakesh0x commented:
https://github.com/user-attachments/assets/cc677de7-5ab1-4953-aecb-f2b402b379f0


hey @0xnirmal if any changes is need, pls let me know, this is half build will be done in within a week or so

0xnirmal commented:
> quickbook_demo.mp4 
> hey @0xnirmal if any changes is need, pls let me know, this is half build will be done in within a week or so

Looks great - one thing to just consider is I tried running `dojo.getState()` in the UI and errored out. Are the dojo hooks properly configured across the entire project here? For ex, try running the Linear one locally and observe the outputs of `dojo.getState()` and `dojo.setState()`

rakesh0x committed:
dojo-hooks-fixes

rakesh0x commented:
> > quickbook_demo.mp4
> > hey @0xnirmal if any changes is need, pls let me know, this is half build will be done in within a week or so
> 
> Looks great - one thing to just consider is I tried running `dojo.getState()` in the UI and errored out. Are the dojo hooks properly configured across the entire project here? For ex, try running the Linear one locally and observe the outputs of `dojo.getState()` and `dojo.setState()`

thanx for the review , fixed this error :)

rakesh0x committed:
customer-reports

rakesh0x committed:
ui-fixes

rakesh0x committed:
ui-fixes

rakesh0x committed:
income-tracker

rakesh0x committed:
invoice

rakesh0x committed:
vendor

rakesh0x committed:
fixes

rakesh0x commented:
https://github.com/user-attachments/assets/1625b061-d018-4894-97cd-e3ee7e49abfb


review needed sir @0xnirmal, would love to do more changes into it

0xnirmal commented:
@Dzaka-athif - can you review this and once completed, get it added to the community section as the first contribution? thank you!

rakesh0x committed:
build-fixes

Dzaka-athif commented:
@rakesh0x hey, can you use [vite-plugin-singlefile](https://www.npmjs.com/package/vite-plugin-singlefile)?  It's going to make it way easier to add your app to the dojo community section. 

Ideally we want the css and the js are contained in the single .html file. 
<img width="235" height="136" alt="Screenshot 2025-11-07 at 7 56 31 am" src="https://github.com/user-attachments/assets/6899e583-3721-4ee7-bc55-4f1190d4fe50" />

Can you also change the way you store the initial data from .json file to just .ts const file and import it. Otherwise we need to put create a folder called src/data/ and put the .json file there to make it run. 
<img width="1017" height="411" alt="Screenshot 2025-11-07 at 7 57 14 am" src="https://github.com/user-attachments/assets/4db2146c-afde-4440-af1b-409bf8af24b3" />

rakesh0x committed:
converted json to ts file and build in .html

rakesh0x commented:
hey @Dzaka-athif , I have updated the code, now its using vite-plugin-singlefile and builds are now in a single .html file 

<img width="329" height="163" alt="image" src="https://github.com/user-attachments/assets/acaaa863-f8f2-4a0a-a3f8-87206f8cb73a" />

i have also converted those JSON files into .ts file in the following src/data directory 

<img width="329" height="163" alt="image" src="https://github.com/user-attachments/assets/f796d4a0-9aae-4cbb-b3e8-4d915c3cbd79" />

Dzaka-athif reviewed (approved):
lgtm!

rakesh0x committed:
Merge remote-tracking branch 'upstream/main' into quickbooks
